troubleshoot/elasticsearch/high-cpu-usage.md

```diff
@@ -77,17 +75,54 @@ This API returns a breakdown of any hot threads in plain text. High CPU usage fr
+
+For details on diagnosing and resolving these issues, see [hot spotting](hotspotting.md).
+
+**Oversharding**
+
+If your Elasticsearch cluster contains a large number of shards, you might be facing an oversharding issue.
+
+Oversharding occurs when there are too many shards, causing each shard to be smaller than optimal. While Elasticsearch doesn’t have a strict minimum shard size, an excessive number of small shards can negatively impact performance. Each shard consumes cluster resources since Elasticsearch must maintain metadata and manage shard states across all nodes.
+
+If you have too many small shards, you can address this by:
 
-Heavy indexing and search loads can deplete smaller thread pools. To better handle heavy workloads, add more nodes to your cluster or upgrade your existing nodes to increase capacity.
+* Removing empty or unused indices.
+* Deleting or closing indices containing outdated or unnecessary data.
+* Reindexing smaller shards into fewer, larger shards to optimize cluster performance.
 
-**Spread out bulk requests**
+See [Size your shards](/deploy-manage/production-guidance/optimize-performance/size-shards.md) for more information.
 
-While more efficient than individual requests, large [bulk indexing](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-bulk) or [multi-search](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-msearch) requests still require CPU resources. If possible, submit smaller requests and allow more time between them.
+### Additional recommendations
 
-**Cancel long-running searches**
+To further reduce CPU load or mitigate temporary spikes in resource usage, consider these steps:
 
-Long-running searches can block threads in the `search` thread pool. To check for these searches, use the [task management API](https://www.elastic.co/docs/api/doc/elasticsearch/group/endpoint-tasks).
+* Scale your cluster: Heavy indexing and search loads can deplete smaller thread pools.Add nodes or upgrade existing ones to handle increased indexing and search loads more effectively.
+* Spread out bulk requests: Submit smaller [bulk indexing](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-bulk-1) or multi-search requests and space them out to avoid overwhelming thread pools.
+* Cancel long-running searches: Regularly use the task management API to identify and cancel searches that consume excessive CPU time. To check
+for these searches, use the [task management API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-tasks-list).
```

Comment:
```suggestion
* **Spread out bulk requests**: Submit smaller [bulk indexing](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-bulk-1) or multi-search requests, and space them out to avoid overwhelming thread pools.
* **Cancel long-running searches**: Regularly use the [task management API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-tasks-list) to identify and cancel searches that consume excessive CPU time.